Dx 663/claim only flag

W3stside opened the pull request:
environment flag for creating builds ONLY for slow.trade frontend w/limited functionality.

When DutchX upgrades, users can have stuck funds in older contracts. This is to prevent this.

TODO:
 - [x] Set contract versions to use

To test: `npm run start:dev:readOnly`

W3stside committed:
added in var plus local test script

W3stside committed:
updated Router to render a diff app on CLAIM_ONLY + stylings

W3stside committed:
added noMenu to Header (for CLAIM_ONLY)

W3stside committed:
Creation of ClaimOnly component

1. used in Home component with some logic restructuring

W3stside committed:
TextSquare: claimOnly flag to remove certain elements on claim_only

W3stside committed:
show banner on env CLAIM_ONLY

W3stside committed:
remove clickable tr from MenuAuctions

1. impeding flow of CLAIM_ONLY functionality when clicking claim

W3stside committed:
added globals (URLS)

W3stside committed:
added proper scrips for building readOnly

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

Velenir commented on src/components/MenuAuctions/index.tsx:
```diff
@@ -72,8 +72,13 @@ export class MenuAuctions extends React.Component <MenuAuctionProps> {
+                        <tr>
+                          <td
+                            className="sectionLink"
+                            onClick={() => push(`/auction/${auction.sell.symbol}-${auction.buy.symbol}-${auction.current.index}`)}
```
Why move onClick to `<td>`? Just because?

W3stside replied:
When clicking claim it would trigger the location change

Velenir replied:
makes sense :+1:

Velenir commented on src/router.tsx:
```diff
@@ -31,7 +31,7 @@ interface AppRouterProps {
 
 // TODO: consider redirecting from inside /order, /wallet, /auction/:nonexistent_addr to root
-const withHeaderAndFooter = (Component: React.ComponentClass | React.SFC, headerProps?: { content?: boolean, dumb?: boolean }, useFooter = true) => (compProps: any) => (
+const withHeaderAndFooter = (Component: React.ComponentClass | React.SFC, headerProps?: { content?: boolean, dumb?: boolean, noMenu?: boolean }, useFooter = true, compProps?: any) => () => (
```
Return `compProps` to the curried function params. Otherwise all props on eg. `HomeWH` disappear into the void. Like router props.
You can use `extraProps` or something instead

W3stside replied:
good catch

Velenir reviewed (changes requested):
Great work!

aside from `Set contract versions to use` we'll also need `build:prod:readOnly` in travis config

W3stside committed:
request changes + others

1. add back curried return of compProps
2. undo testing reverse boolean in TextSquare component
3. re-add clickable td in MenuAuctions unless claimable

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

W3stside committed:
READ_ONLY >> CLAIM_ONLY nomenclature

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

W3stside committed:
fixed null slectoredProvider prop check

W3stside committed:
re-establish verification

W3stside committed:
relative positioning warning banner

W3stside committed:
logic simplification in AppValidator for banners

1. added Development warning for coding ease

W3stside committed:
created logic for grabbing old contracts in contracts api

1. uses localForage

W3stside committed:
DXContractPicker Compnent birth

1. allows choosing and reloading of app to use contract version chooser
2. localForage

W3stside committed:
created localProd claimOnly script + test networksOld

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

W3stside committed:
styling

W3stside committed:
more styles

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

W3stside committed:
logic fix incase list changes, check each time

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

Velenir commented on src/components/DXContractSwitcher/index.tsx:
```diff
@@ -0,0 +1,101 @@
+    this.setState({
+      VERSION_TO_USE: version,
+      CONTRACT_ADDRESSES_TO_USE: OLD_CONTRACT_OBJECT_TO_USE,
+      DX_ADDRESS_TO_USE: OLD_CONTRACT_OBJECT_TO_USE['DutchExchangeProxy'][network].address,
```
This will trigger change in `./contracts.ts` later on, right?

Velenir commented on src/api/contracts.ts:
```diff
@@ -125,6 +125,43 @@ if (process.env.FE_CONDITIONAL_ENV === 'development') {
+  }
+
+  grabOldDXNetworksAndSet()
+  .then((networks: any) => {
```
This whole block and especially this function needs to be run before `contr.deployed()`, better yet before `.setProvider()`. Otherwise we'll have a race condition and who knows which addresses will get applied.

First line in `function init()` is a good place

W3stside replied:
fixed, cheers

W3stside committed:
Merge branch 'develop' into DX-663/Claim-Only-Flag

W3stside committed:
fixed async logic

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

W3stside committed:
Merge branch 'develop' into DX-663/Claim-Only-Flag

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr425--dxreact.review.gnosisdev.com

anxolin reviewed (approved):
Thanks David

Can we deploy this in a staging server?

W3stside committed:
Updated to release2.0 spec + linting

1. package.json:
    a. use latest dx-contracts
    b. use npx ... for truffle cmds
2. update .sol files to pragme ^0.4.24
3. FE - indicate what uses a central server

W3stside committed:
update contracts api for release 2.0

W3stside commented:
closing in favour of #443
